File: compare_the_triplets/compare_the_triplets.rb
def compareTheTriplets(a, b)
  puts [get_score(a, b), get_score(b, a)]
end

def add_point(a, b)
  a > b ? 1 : 0
end

def get_score(a, b)
  a.zip(b).map do |first, second|
    add_point(first, second)
  end.reduce(0, :+)
end

a = [5, 6, 7]
b = [3, 6, 10]

compareTheTriplets(a, b)

File: compare_the_triplets/compare_the_triplets.js
function getScore(a, b) {
  let zipped = zip(a, b);
  let add = (x, y) => x + y;
  return zipped.map(scorePoint).reduce(add);
}

function scorePoint(arr) {
  if (arr[0] > arr[1]) { return 1; }
  return 0;
}

function zip(a, b) {
  let final = [];
  for (let i = 0; i < a.length; i++) {
    final[i] = [a[i], b[i]];
  }
  return final;
}

a = [5, 6, 7];
b = [3, 6, 10];

function compareTriplets(a, b) {
  return [getScore(a, b), getScore(b, a)];
}

console.log(compareTriplets(a, b))
